Backup sources docs, second try (#3288)
conan-io/docs: reference/backup_sources.rst

.. _conan_backup_sources:

Backup sources
==============

The *backup sources* feature is intended for storing the downloaded recipe sources in a secondary location,

> allowing future reproducibility of your builds even in the case where the original mirrors are no longer accessible.

Reviewer: ```suggestion
allowing future reproducibility of your builds even in the case where the original download URLs are no longer accessible.
```

The backup is triggered for the :ref:`download<conan_tools_files_get>` and :ref:`get<conan_tools_files_get>` methods
when a ``sha256`` signature is provided.

Configuration
-------------
This feature is controlled by a few configuration items:

core.sources:download_cache
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Local path to store the sources backup to.

Reviewer: (optional), if not defined it will store things in the ``.conan2/`` folder

> Local path to store the sources backup to.

Author: ```suggestion
(Optional) Local path to store the sources backup to. If not set, the default Conan home cache path will be used.
```

core.sources:download_urls
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
List of URLs that Conan will try to download the sources from, where ``origin`` represents the original URL.
This allows to control the fetch order, either ``["origin", "https://your.backup/remote/"]``
to look into and fetch from your backup remote only if and when the original source is not present,
or ``["https://your.backup/remote/", "origin"]`` to prefer your remote ahead of the recipes.
(*Note that being a list, multiple remotes are also possible*)

core.sources:upload_url
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> URL of the remote to upload the backups when calling ``conan upload``,

Author: ```suggestion
(Optional) URL of the remote to upload the backups when calling ``conan upload``,
```

> which might or might be not different from any of the URLs defined for download.

Reviewer: ```suggestion
which might or might not be different from any of the URLs defined for download.
```
